### PR TITLE
feat(influxdb): Prevent the deletion of system buckets

### LINKED ISSUE #15630: protect system buckets from deletion
If a user attempts to delete a system bucket over the API, the request should be rejected. We need to continue to support internal deletions of system buckets, as cleanup when an organization is deleted.

### PR DESCRIPTION
**Pull request by bthesorceror**
Closes #15630

Keep users from removing system buckets.

### CODE REVIEW

File: kv/bucket_test.go [hunk 1]
```diff
@@ -17,6 +22,75 @@ func TestInmemBucketService(t *testing.T) {
 	influxdbtesting.BucketService(initInmemBucketService, t)
 }
 
+func TestSystemBucketDeletion(t *testing.T) {
```
Comment (jademcgough) [inline kv/bucket_test.go]: Is there a reason to have these tests in here, and not in `testing/bucket_service.go`? I'm not super worried about functionality diverging between different kv implementations, but it's a little weird that these only run against bolt.

Comment (bthesorceror) [reply kv/bucket_test.go]: the http bucket service doesn't allow for creating buckets with a type of system and it also uses that test group.

Comment (bthesorceror) [reply kv/bucket_test.go]: I suppose the question is, should the http and kv levels be running the same tests?

File: kv/bucket.go [hunk 2]
```diff
@@ -683,7 +683,7 @@ func (s *Service) updateBucket(ctx context.Context, tx Tx, id influxdb.ID, upd i
 		b.Description = *upd.Description
 	}
 
-	if upd.Name != nil {
+	if upd.Name != nil && b.Type != influxdb.BucketTypeSystem {
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply kv/bucket.go]: 👍

File: kv/bucket_test.go (hunk 1 above)
Comment (jademcgough) [reply kv/bucket_test.go]: `http/bucket_service.go` is different from `testing/bucket_service.go`. The former is only through the http API, the latter is unit testing of public service functions and runs against bolt, inmem and etcd.

Comment (kelwang) [reply kv/bucket_test.go]: I think initially we have the same tests in http level to save some code. But recently, it seems http level has different behaviors, so i think it's not that necessary now

**Pull request merged**